src/StreamJsonRpc/CommonMethodNameTransforms.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using Microsoft;
+
+    /// <summary>
+    /// Common RPC method transform functions that may be supplied to <see cref="JsonRpc.AddLocalRpcTarget(object, System.Func{string, string})"/>.
+    /// </summary>
+    public static class CommonMethodNameTransforms
+    {
+        /// <summary>
+        /// Gets a function that converts a given string's first character to lowercase.
+        /// </summary>
+        public static Func<string, string> CamelCase
+        {
+            get
+            {
+                return name =>
+                {
+                    if (name == null)
+                    {
+                        throw new ArgumentNullException();
+                    }
+
+                    if (name.Length == 0)
+                    {
+                        return string.Empty;
+                    }
+
+                    return name.Substring(0, 1).ToLowerInvariant() + name.Substring(1);
```

Comment:
Because you're doing some allocations here, you may want to check if the first character is already lowercase. That avoids 3 allocations if it is.

Reply:
It's also worth noting that your algorithm for camelCase is different than what NewtonSoft uses with the camelCase converter. The main difference that I see is that the NewtonSoft version would lowercase all uppercase characters until the last one before a lowercase. So, for example, "HELLOWorld" would become "helloWorld" (instead of "hELLLOWorld"). Also, any fully uppercase string would be fully lowercase (i.e. "HELLO" would become "hello" instead of "hELLO").

Reply:
Regarding the allocations, you're right. However I considered that making the logic more complex with branching would add test cost and in the end the allocations would be insignificant. This function is only executed once per RPC method definition, not once per call, so it's a fixed cost, which is actually quite small relative to even one call at runtime. 

Interesting point about the camel casing. Certainly any changes to how this function works would result in a wire protocol breaking change, so whatever we check in will be locked in forever. I'll see if I can reuse Newtonsoft's implementation since they have more experience in this area than I do. Thanks for sharing.